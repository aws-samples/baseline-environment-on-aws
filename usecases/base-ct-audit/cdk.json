{
  "app": "npx ts-node --prefer-ts-exts bin/blea-base-ct-audit.ts",
  "context": {
    "dev-audit": {
      "description": "Context samples for ControlTower Audit Account - Specific account & region",
      "env": {
        "account": "333333333333",
        "region": "ap-northeast-1"
      },
      "slackNotifier": {
        "workspaceId": "T8XXXXXXX",
        "channelIdAgg": "C01ZZZZZZZZ"
      }
    }
  }
}
