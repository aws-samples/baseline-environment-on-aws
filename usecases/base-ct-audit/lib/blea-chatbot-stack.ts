import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { aws_chatbot as cb } from 'aws-cdk-lib';
import { aws_iam as iam } from 'aws-cdk-lib';

export interface BLEAChatbotStackProps extends cdk.StackProps {
  topicArn: string;
  channelId: string;
  workspaceId: string;
}

// NOTICE: AWS Chatbot can send events from supported services only.
// See: https://docs.aws.amazon.com/chatbot/latest/adminguide/related-services.html
export class BLEAChatbotStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props: BLEAChatbotStackProps) {
    super(scope, id, props);

    // AWS Chatbot configuration for sending message
    const chatbotRole = new iam.Role(this, 'ChatbotRole', {
      assumedBy: new iam.ServicePrincipal('chatbot.amazonaws.com'),
      managedPolicies: [
        iam.ManagedPolicy.fromAwsManagedPolicyName('ReadOnlyAccess'),
        iam.ManagedPolicy.fromAwsManagedPolicyName('CloudWatchReadOnlyAccess'),
      ],
    });

    // !!! Create SlackChannel and add aws chatbot app to the room
    new cb.CfnSlackChannelConfiguration(this, 'ChatbotChannel', {
      configurationName: `${id}-${props.workspaceId}`,
      slackChannelId: props.channelId,
      iamRoleArn: chatbotRole.roleArn,
      slackWorkspaceId: props.workspaceId,
      snsTopicArns: [props.topicArn],
    });
  }
}
