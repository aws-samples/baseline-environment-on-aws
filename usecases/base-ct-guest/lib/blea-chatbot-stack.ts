import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { aws_chatbot as cb } from 'aws-cdk-lib';
import { aws_iam as iam } from 'aws-cdk-lib';
import { aws_sns as sns } from 'aws-cdk-lib';

export interface BLEAChatbotStackProps extends cdk.StackProps {
  topicArn: string;
  channelId: string;
  workspaceId: string;
}

// NOTICE: AWS Chatbot can send events from supported services only.
// See: https://docs.aws.amazon.com/ja_jp/chatbot/latest/adminguide/related-services.html
export class BLEAChatbotStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props: BLEAChatbotStackProps) {
    super(scope, id, props);

    // !!! Create SlackChannel and add aws chatbot app to the room
    const slackChatbotChannel = new cb.SlackChannelConfiguration(this, 'ChatbotChannel', {
      slackChannelConfigurationName: `${id}-${props.workspaceId}`,
      slackChannelId: props.channelId,
      slackWorkspaceId: props.workspaceId,
    });

    // AWS Chatbot configuration for sending message
    slackChatbotChannel.addNotificationTopic(sns.Topic.fromTopicArn(this, 'ChatbotTopic', props.topicArn));
    slackChatbotChannel.role?.addManagedPolicy(iam.ManagedPolicy.fromAwsManagedPolicyName('ReadOnlyAccess'));
  }
}
