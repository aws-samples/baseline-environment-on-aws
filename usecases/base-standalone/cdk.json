{
  "app": "npx ts-node --prefer-ts-exts bin/blea-base-sa.ts",
  "context": {
    "dev": {
      "description": "Context samples for Dev - Anonymous account & region",
      "envName": "Development",
      "securityNotifyEmail": "notify-security@example.com",
      "slackNotifier": {
        "workspaceId": "T8XXXXXXX",
        "channelIdSec": "C00XXXXXXXX"
      }
    },
    "stage": {
      "description": "Context samples for Staging - Specific account & region  ",
      "env": {
        "account": "111111111111",
        "region": "ap-northeast-1"
      },
      "envName": "Staging",
      "securityNotifyEmail": "notify-security@example.com",
      "slackNotifier": {
        "workspaceId": "T8XXXXXXX",
        "channelIdSec": "C01XXXXXXXX"
      }
    },
    "prod": {
      "description": "Context samples for Production - Specific account & region  ",
      "env": {
        "account": "222222222222",
        "region": "ap-northeast-1"
      },
      "envName": "Production",
      "securityNotifyEmail": "notify-security@example.com",
      "slackNotifier": {
        "workspaceId": "T8XXXXXXX",
        "channelIdSec": "C02XXXXXXXX"
      }
    }
  }
}
