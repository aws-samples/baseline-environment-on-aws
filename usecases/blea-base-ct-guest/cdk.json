{
  "app": "npx ts-node --prefer-ts-exts bin/blea-base-ct-guest.ts"
}
