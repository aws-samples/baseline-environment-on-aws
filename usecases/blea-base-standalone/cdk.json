{
  "app": "npx ts-node --prefer-ts-exts bin/blea-base-standalone.ts"
}
