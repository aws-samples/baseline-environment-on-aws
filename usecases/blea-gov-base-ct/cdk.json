{
  "app": "npx ts-node --prefer-ts-exts bin/blea-gov-base-ct.ts",
  "context": {
    "@aws-cdk/aws-s3:serverAccessLogsUseBucketPolicy": true
  }
}
