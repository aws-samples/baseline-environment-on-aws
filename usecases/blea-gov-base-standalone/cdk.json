{
  "app": "npx ts-node --prefer-ts-exts bin/blea-gov-base-standalone.ts",
  "context": {
    "@aws-cdk/aws-s3:serverAccessLogsUseBucketPolicy": true
  }
}
