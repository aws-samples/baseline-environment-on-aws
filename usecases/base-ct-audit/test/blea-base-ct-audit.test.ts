import * as cdk from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';

import { BLEAChatbotStack } from '../lib/blea-chatbot-stack';

// Load cdk.json to get context parameters
import * as cdk_json from '../cdk.json';

const pjPrefix = 'BLEA';
const app = new cdk.App();
const envKey = 'dev-audit';
const envVals = cdk_json['context'][envKey];

describe(`${pjPrefix} ControlTower Stacks`, () => {
  test('AuditAccount Stacks', () => {
    const procEnv = envVals['env']; // contains "account" and "region"
    const workspaceId = envVals['slackNotifier']['workspaceId'];
    const channelIdAggregate = envVals['slackNotifier']['channelIdAgg'];
    const controltowerAuditAccountId = procEnv['account'];
    const controltowerHomeRegion = procEnv['region'];

    const aggregateTopicArn = `arn:aws:sns:${controltowerHomeRegion}:${controltowerAuditAccountId}:aws-controltower-AggregateSecurityNotifications`;
    const chatbotAggregate = new BLEAChatbotStack(app, `${pjPrefix}-ChatbotAggregate`, {
      topicArn: aggregateTopicArn,
      workspaceId: workspaceId,
      channelId: channelIdAggregate,
      env: procEnv,
    });

    // test with snapshot
    expect(Template.fromStack(chatbotAggregate)).toMatchSnapshot();
  });
});
