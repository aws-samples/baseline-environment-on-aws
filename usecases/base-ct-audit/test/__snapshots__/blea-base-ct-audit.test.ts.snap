// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BLEA ControlTower Stacks AuditAccount Stacks 1`] = `
Object {
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "ChatbotChannel": Object {
      "Properties": Object {
        "ConfigurationName": "BLEA-ChatbotAggregate-T8XXXXXXX",
        "IamRoleArn": Object {
          "Fn::GetAtt": Array [
            "ChatbotRole8A87AA1F",
            "Arn",
          ],
        },
        "SlackChannelId": "C01ZZZZZZZZ",
        "SlackWorkspaceId": "T8XXXXXXX",
        "SnsTopicArns": Array [
          "arn:aws:sns:ap-northeast-1:333333333333:aws-controltower-AggregateSecurityNotifications",
        ],
      },
      "Type": "AWS::Chatbot::SlackChannelConfiguration",
    },
    "ChatbotRole8A87AA1F": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "chatbot.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/ReadOnlyAccess",
              ],
            ],
          },
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/CloudWatchReadOnlyAccess",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
